taxonomy/algolia/utils.py

```diff
@@ -41,6 +44,83 @@ def index_jobs_data_in_algolia():
     client.replace_all_objects(jobs_data)
     LOGGER.info('[TAXONOMY] Jobs data successfully indexed on algolia.')
 
+def calculate_jaccard_similarity(set_a, set_b):
+    try:
+        intersection = set_a.intersection(set_b)
+        jaccard_similarity = len(intersection)/len(set_a.union(set_b))
+        return jaccard_similarity
+    except:
+        return 0
+
+def combine_jobs_and_skills_data(jobs):
+    """
+    Combine jobs and skills data.
+
+    Returns:
+        (list<dict>): A list of dicts containing job and their skills in a list.
+    """
+    jobs = jobs.all()
+
+    all_job_and_skills_data = []
+    for job in jobs:
+        job_skills = JobSkills.objects.filter(job=job)
+        skills = []
+        for job_skill in job_skills:
+            skills.append(job_skill.skill.name)
+
+        # Append job and skills in a dictionary to the list
+        all_job_and_skills_data.append({
+            'name': job.name,
+            'skills': skills,
+        })
+
+    return all_job_and_skills_data
+
+def calculate_job_recommendations(all_jobs_and_skills):
+    """
+    Calculate job recommendations.
+
+    Args:
+        job (dict): A dict containing job data.
+        all_jobs_and_skills (list<dict>): A list of dicts containing job and their skills in a list.
+
+    Returns:
+        (list<dict>): A list of dicts containing jobs and their recommended jobs.
+    """
+    jobs = []
```

Comment:
what is this list used for?

Reply:
It is renamed to `candidate_jobs`. It is being used to create a data structure which is being used in creating a dataframe where this list is being used to contain repeated job names and the other list `matching_jobs` is being used to store job names mapped to the job names in this list. We have another list named `jaccard_similarities` storing similarity scores between a pair of jobs at an index. For example

candidate_jobs[7] = 'Job A'
matching_jobs[7] = 'Job B'
jaccard_similarities[7] = <similarity score between Job A and Job B>